src/System.Linq/src/System/Linq/Partition.cs

```diff
@@ -35,16 +35,45 @@ internal interface IIListProvider<TElement> : IEnumerable<TElement>
         int GetCount(bool onlyIfCheap);
     }
 
+    /// <summary>
+    /// An iterator that supports random access and can produce a partial sequence of its items through an optimized path.
+    /// </summary>
     internal interface IPartition<TElement> : IIListProvider<TElement>
     {
+        /// <summary>
+        /// Skips the specified number of elements in this sequence.
+        /// </summary>
+        /// <param name="count">The number of elements to skip.</param>
+        /// <returns>An <see cref="IPartition{TElement}"/> with the first <paramref name="count"/> items removed.</returns>
         IPartition<TElement> Skip(int count);
 
+        /// <summary>
+        /// Takes the specified number of elements from this sequence.
+        /// </summary>
+        /// <param name="count">The number of elements to take.</param>
+        /// <returns>An <see cref="IPartition{TElement}"/> with only the first <paramref name="count"/> items.</returns>
         IPartition<TElement> Take(int count);
 
+        /// <summary>
+        /// Gets the item associated with a 0-based index in this sequence.
+        /// </summary>
+        /// <param name="index">The 0-based index to access.</param>
+        /// <param name="found"><c>true</c> if the sequence contains an element at that index, <c>false</c> otherwise.</param>
+        /// <returns>The element if <paramref name="found"/> is <c>true</c>, otherwise, the default value of <see cref="TElement"/>.</returns>
         TElement TryGetElementAt(int index, out bool found);
 
+        /// <summary>
+        /// Gets the first item in this sequence.
+        /// </summary>
+        /// <param name="found"><c>true</c> if the sequence contains a first element, <c>false</c> otherwise.</param>
```

Comment:
"contains a first element"... could this just be "contains an element"?  Same for TryGetLast.